### PR TITLE
add ProductCatalogPushTemplate

### PR DESCRIPTION
**Pull request by rymorale**
## Description
- add ProductCatalogPushTemplate class
- also cleanup unused constants

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_LAYOUT}\" not found.")
+        rawCatalogItems = DataReader.optString(
+            data, PushTemplateConstants.PushPayloadKeys.CATALOG_ITEMS, null
+        )
+            ?: throw IllegalArgumentException("Required field \"${PushTemplateConstants.PushPayloadKeys.CATALOG_ITEMS}\" not found.")
+        catalogItems = parseCatalogItemsFromString(rawCatalogItems)
+    }
+
+    constructor(intent: Intent) : super(intent) {
+        intentAction = intent.action
```
Comment (spoorthipujariadobe) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:85]: Can we use `ctaButtonUri` instead?

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+import android.content.Intent
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.util.DataReader
+import org.json.JSONArray
+import org.json.JSONException
+
+internal class ProductCatalogPushTemplate : AEPPushTemplate {
+    internal var intentAction: String? = null
```
Comment (spoorthipujariadobe) [inline code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:22]: Is this same as `ctaButtonUri`? Don't see it assigned in the map constructor

Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:22]: this was a copy/paste error... removed it

**Review by prudrabhat — CHANGES_REQUESTED** [review]

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        private fun parseCatalogItemsFromString(catalogItemsString: String?): MutableList<CatalogItem> {
+            val catalogItems = mutableListOf<CatalogItem>()
+            var jsonArray: JSONArray? = null
+            try {
+                jsonArray = JSONArray(catalogItemsString)
+            } catch (e: JSONException) {
+                Log.error(
+                    PushTemplateConstants.LOG_TAG,
+                    SELF_TAG,
+                    "Exception occurred when creating json array from the catalog items string: ${e.localizedMessage}"
```
[the remark this answers is not in the document]
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:113]: i'll get this fixed

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+        // Required, Price of this product to display when the notification is selected
+        val price: String,
+
+        // Required, URI to be handled when the user clicks the large image of the selected item
+        val uri: String
+    )
+
+    constructor(data: Map<String, String>) : super(data) {
+        ctaButtonText = DataReader.optString(
```
[the remark this answers is not in the document]
Comment (prudrabhat) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:68]: We can discuss and add it later too.

**Review by prudrabhat — APPROVED** [review]

File: code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (rymorale) [reply code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/templates/ProductCatalogPushTemplate.kt:68]: i like this idea. I have pr coming soon that adds the ProductCatalogNotificationBuilder and can add these changes there.

**Pull request merged**